provider/core/src/request.rs

```diff
@@ -298,81 +312,115 @@ impl fmt::Debug for DataLocale {
     }
 }
 
-impl Writeable for DataLocale {
-    fn write_to<W: core::fmt::Write + ?Sized>(&self, sink: &mut W) -> core::fmt::Result {
-        self.langid.write_to(sink)?;
-        if !self.keywords.is_empty() {
-            sink.write_str("-u-")?;
-            self.keywords.write_to(sink)?;
-        }
-        Ok(())
-    }
-
-    fn writeable_length_hint(&self) -> LengthHint {
-        let mut length_hint = self.langid.writeable_length_hint();
-        if !self.keywords.is_empty() {
-            length_hint += self.keywords.writeable_length_hint() + 3;
-        }
-        length_hint
-    }
-
-    fn write_to_string(&self) -> alloc::borrow::Cow<str> {
-        if self.keywords.is_empty() {
-            return self.langid.write_to_string();
-        }
-        let mut string =
-            alloc::string::String::with_capacity(self.writeable_length_hint().capacity());
-        let _ = self.write_to(&mut string);
-        alloc::borrow::Cow::Owned(string)
-    }
-}
-
-writeable::impl_display_with_writeable!(DataLocale);
+icu_locale_core::impl_writeable_for_each_subtag_str_no_test!(DataLocale, selff, selff.script.is_none() && selff.region.is_none() && selff.variant.is_none() && selff.subdivision.is_none() => selff.language.write_to_string());
```

Comment:
Nit optional: I think I prefer the explicit manual impls than calling this internal macro at a distance